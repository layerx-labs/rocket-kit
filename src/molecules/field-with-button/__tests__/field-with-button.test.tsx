import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { FieldWidthButton } from '../../..';

describe('FieldWidthButton', () => {
  it('renders', () => {
    const { asFragment } = render(
      <FieldWidthButton type="text" value="this is my text" buttonIcon="copy" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('has correct value', () => {
    const textValue = 'this is my text';
    render(
      <FieldWidthButton type="text" value={textValue} buttonIcon="copy" />
    );
    const foundInput = screen.getByDisplayValue(textValue);
    expect(foundInput).toHaveValue(textValue);
  });

  it('calls action callback', async () => {
    const onClickAction = jest.fn();
    const textValue = 'https://taikai.network';
    const buttonText = 'Copy';

    render(
      <FieldWidthButton
        type="text"
        value={textValue}
        buttonIcon="copy"
        buttonValue={buttonText}
        buttonAction={onClickAction}
      />
    );

    const button = screen.getByRole('button', { name: buttonText });
    userEvent.click(button);
    expect(onClickAction).toBeCalledTimes(1);
  });
});
