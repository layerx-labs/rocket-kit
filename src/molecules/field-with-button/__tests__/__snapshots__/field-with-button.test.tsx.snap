// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FieldWidthButton renders 1`] = `
<DocumentFragment>
  .c3 {
  --button: #56cad7;
  --txt: #ffffff;
  --hover: #2ba8b6;
  --button: #56cad7;
  --hover: #2ba8b6;
  border-width: 0;
  border-style: solid;
  border-color: var(--button);
  border-radius: 999px;
  background-color: var(--button);
  width: 2.25rem;
  height: 2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c3:hover {
  background-color: var(--hover);
}

.c3:disabled {
  cursor: inherit;
  opacity: 0.5;
}

.c3:disabled:hover {
  pointer-events: none;
}

.c3 > *:not(:last-child) {
  margin-left: 0;
  margin-right: 5px;
}

.c3 span {
  position: relative;
  font-size: 0.75rem;
  font-weight: 500;
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  color: var(--txt);
  pointer-events: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  -webkit-order: 2;
  -ms-flex-order: 2;
  order: 2;
}

.c3 svg {
  -webkit-order: 1;
  -ms-flex-order: 1;
  order: 1;
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: var(--txt);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c2 {
  border-width: 0.0625rem;
  border-style: solid;
  border-color: #939393;
  border-radius: 6px;
  height: 3.125rem;
  padding: 0.625rem;
  font-family: inherit;
  font-size: 1rem;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c2:focus {
  border-color: #4329a6;
  outline: none;
}

.c2:disabled {
  background-color: #c4c7d3;
  color: #939393;
}

.c2:invalid {
  border-color: #ef5766;
  outline: none;
  box-shadow: none;
  -webkit-box-shadow: none;
  -moz-box-shadow: none;
}

.c2:required {
  border-color: #939393;
}

.c2::-webkit-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c2::-moz-placeholder {
  color: #939393;
  opacity: 1;
}

.c2:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c2::placeholder {
  color: #939393;
  opacity: 1;
}

.c2:-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c2::-ms-input-placeholder {
  color: #939393;
  opacity: 1;
}

.c0 label {
  margin-bottom: 0.3125rem;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 input {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 0.375rem 0 0 0.375rem;
}

.c1 input:focus {
  outline: none;
}

.c1 input:focus + button {
  background-color: #4329a6;
}

.c1 span {
  -webkit-order: 3;
  -ms-flex-order: 3;
  order: 3;
  width: 100%;
}

.c1 button {
  border: 0;
  border-radius: 0 0.375rem 0.375rem 0;
  background-color: #939393;
  min-width: 3.125rem;
  height: 3.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.25rem;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c1 button svg {
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: #ffffff;
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c1 button:hover {
  background-color: #7a7a7a;
}

.c1 button:disabled {
  cursor: inherit;
  opacity: 0.5;
}

.c1 button:disabled:hover {
  background-color: #939393;
  pointer-events: none;
}

@media (min-width:1280px) {
  .c3 {
    min-width: 2.25rem;
  }
}

<div
    class="c0"
  >
    <div
      class="c1"
    >
      <input
        class="c2 text-field"
        placeholder=""
        type="text"
        value="this is my text"
      />
      <button
        class="c3 button"
        color="green"
        value=""
      >
        <svg
          class="icon"
          viewBox="0 0 32 32"
        >
          <path
            d="M20 1.333h-14.667c-1.467 0-2.667 1.2-2.667 2.667v17.333c0 0.733 0.6 1.333 1.333 1.333s1.333-0.6 1.333-1.333v-16c0-0.733 0.6-1.333 1.333-1.333h13.333c0.733 0 1.333-0.6 1.333-1.333s-0.6-1.333-1.333-1.333zM25.333 6.667h-14.667c-1.467 0-2.667 1.2-2.667 2.667v18.667c0 1.467 1.2 2.667 2.667 2.667h14.667c1.467 0 2.667-1.2 2.667-2.667v-18.667c0-1.467-1.2-2.667-2.667-2.667zM24 28h-12c-0.733 0-1.333-0.6-1.333-1.333v-16c0-0.733 0.6-1.333 1.333-1.333h12c0.733 0 1.333 0.6 1.333 1.333v16c0 0.733-0.6 1.333-1.333 1.333z"
          />
        </svg>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
